llvm/lib/CodeGen/MachinePipeliner.cpp

```diff
@@ -141,7 +141,7 @@ static cl::opt<int> SwpForceII("pipeliner-force-ii",
 static cl::opt<int>
     SwpMaxStages("pipeliner-max-stages",
                  cl::desc("Maximum stages allowed in the generated scheduled."),
-                 cl::Hidden, cl::init(3));
+                 cl::Hidden, cl::init(-1));
```

Comment:
In the end, this option seems strange. This decision should always delegated to the target.

Reply:
On the other hand, we can push this option to -1 from the frontend.

Reply:
I guess this predates shouldUseSchedule. I agree with Andreu that it should be decided there.

Reply:
In fact I will do that in this PR.

Reply:
Hmm. It's more complicated than that, because there's a MaxStages test in the main II loop. That means it could be moved to canAcceptII, but that's not upstreamed either. I propose to upstream both at the same time, mving the MaxStages check to the default implementation of canAcceptII